test/unit/bundles/inspec-compliance/api_test.rb

```diff
@@ -247,32 +247,65 @@
   end
 
   describe '.determine_server_type' do
-    it 'returns `automate` when a 401 is received from `https://automate.example.com/compliance/version`' do
-      good_response = mock
+    let(:url) { 'https://someserver.onthe.net/' }
+
+    let(:compliance_endpoint) { '/api/version' }
+    let(:automate_endpoint) { '/compliance/version' }
+    let(:headers) { nil }
+    let(:insecure) { true }
+
+    let(:good_response) { mock }
+    let(:bad_response) { mock }
+
+    it 'returns `:automate` when a 401 is received from `https://URL/compliance/version`' do
       good_response.stubs(:code).returns('401')
-      url = 'https://automate.example.com'
-      Compliance::HTTP.expects(:get).with(url + '/compliance/version', nil, true).returns(good_response)
-      Compliance::API.determine_server_type(url, true).must_equal(:automate)
+
+      Compliance::HTTP.expects(:get)
+        .with(url + automate_endpoint, headers, insecure)
+        .returns(good_response)
+
+      Compliance::API.determine_server_type(url, insecure).must_equal(:automate)
+    end
+
+    # Chef Automate currently returns 401 for `/compliance/version` but some
+    # versions of OpsWorks Chef Automate return 200 and a Chef Manage page when
+    # unauthenticated requests are received.
+    it 'returns `:automate` when a 200 is received from `https://URL/compliance/version`' do
```

Comment:
Can we add a test that checks that it returns `nil` if we hit the server, get a 200, but do not get the Chef Manage page? That's the other edge case here.

Reply:
Can do! Great suggestion.